### PR TITLE
Add doc for limiting resources by priority

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> For 1.11 Features: set Milestone to 1.11 and Base Branch to release-1.11
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> NOTE: After opening the PR, please *un-check and re-check* the ["Allow edits from maintainers"](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) box so that maintainers can work on your patch and speed up the review process. This is a temporary workaround to address a known issue with GitHub.> 
>

cc @bsalamat

### CODE REVIEW

File: content/en/docs/concepts/policy/resource-quotas.md [hunk 1]
```diff
@@ -183,6 +183,154 @@ The `Terminating`, `NotTerminating`, and `NotBestEffort` scopes restrict a quota
...
 * `requests.memory`
 
+### Resource Quota Per PriorityClass
+
+Pods can be created at a specific [priority](/docs/concepts/configuration/pod-priority-preemption/#pod-priority). If operator wants to control consumption of
+resources by pods according to the priority of the pod, this can be achieved using `scopeSelector` field in the quota spec.
+
+A quota will be matched and consumed only if `scopeSelector` in the Quota spec selects the pod.
+
+**NOTE:** `scopeSelector` is an alpha field and must be feature gate `ResourceQuotaScopeSelectors` must be enabled before using it.
```
Comment (bsalamat) [inline content/en/docs/concepts/policy/resource-quotas.md]: s/must be feature gate/the feature gate/

File: content/en/docs/concepts/policy/resource-quotas.md [hunk 2]
```diff
@@ -303,6 +451,45 @@ hard limits of each namespace according to other signals.
 Note that resource quota divides up aggregate cluster resources, but it creates no
 restrictions around nodes: pods from several namespaces may run on the same node.
 
+## Limit Priority Class consumption by default
+
+It may be desired that pods at a particular priority, let say  "cluster-services", should be allowed in a namespace, if and only if, a matching quota object exists.
```
Comment (bsalamat) [inline content/en/docs/concepts/policy/resource-quotas.md]: s/let say/e.g.,/

**Comment by tengqm** [comment]
This is related to issue #9098

File: content/en/docs/concepts/policy/resource-quotas.md (hunk 1 above)
Comment (tengqm) [reply content/en/docs/concepts/policy/resource-quotas.md]: please also update content/en/docs/reference/command-line-tools-reference/feature-gates.md to make sure the feature gate is listed.

**Comment by tengqm** [comment]
triggered restart of travis and see if the error is a transient one.

File: content/en/docs/concepts/policy/resource-quotas.md [hunk 3]
```diff
@@ -183,6 +183,167 @@ The `Terminating`, `NotTerminating`, and `NotBestEffort` scopes restrict a quota
...
 * `requests.memory`
 
+### Resource Quota Per PriorityClass
+
+Pods can be created at a specific [priority](/docs/concepts/configuration/pod-priority-preemption/#pod-priority). If operator wants to control consumption of
+resources by pods according to the priority of the pod, this can be achieved using `scopeSelector` field in the quota spec.
+
+A quota will be matched and consumed only if `scopeSelector` in the Quota spec selects the pod.
+
+**NOTE:** `scopeSelector` is an alpha field and must be the feature gate `ResourceQuotaScopeSelectors` must be enabled before using it.
```
Comment (bsalamat) [inline content/en/docs/concepts/policy/resource-quotas.md]: Please remove the first "must be".

**Comment by bsalamat** [comment]
Thanks, @vikaschoudhary16!

**Comment by vikaschoudhary16** [comment]
@tengqm ping

**Comment by mistyhacks** [comment]
/hold

Please rebase on `release-1.11`.

**Comment by vikaschoudhary16** [comment]
Moved here https://github.com/kubernetes/website/pull/9291

**Comment by mistyhacks** [comment]
This needs to be rebased on **current** `master`. I will do it this time. In the future, make sure you are working off a fresh `master`.

File: content/en/docs/concepts/policy/resource-quotas.md [hunk 4]
```diff
@@ -1,325 +1,546 @@
...
+apiVersion: apiserver.k8s.io/v1alpha1
+kind: AdmissionConfiguration
+plugins:
+- name: "ResourceQuota"
+  configuration:
+    apiVersion: resourcequota.admission.k8s.io/v1alpha1
+    kind: Configuration
+    limitedResources:
+    - resource: pods
+    matchScopes:
```
Comment (seh) [inline content/en/docs/concepts/policy/resource-quotas.md]: The indentation is off here. The "matchScopes" field should be indented as a sibling of the "resource" field above, per [the `LimitedResource` struct](https://github.com/kubernetes/kubernetes/blob/d4d02762340bfb6ac93afd10884a3d9941b7322f/plugin/pkg/admission/resourcequota/apis/resourcequota/types.go#L38-L72).

Comment (seh) [reply content/en/docs/concepts/policy/resource-quotas.md]: See #12055.

**Pull request merged**